.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,64 @@
+---   # Proof generation benchmarking workflow
+
+name: Benchmark proving
+
+on:
+  schedule:
+    - cron: "0 4 * * *"
+  workflow_dispatch:
+    branches:
+      - "**"
+
+env:
+  BENCHMARK_WITNESS: artifacts/witness_b1000_b1019.json
+
+jobs:
+  benchmark_proving:
+    name: Benchmark proving for representative blocks
+    runs-on: zero-reg
+    timeout-minutes: 900  # Timeout TBD
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Build the project
+        run: |
+          RUSTFLAGS='-C target-cpu=native -Zlinker-features=-lld -Copt-level=3' cargo build --release
+          sudo sysctl kernel.perf_event_paranoid=0
+
+      - name: Download previous results
+        uses: dawidd6/action-download-artifact@v6
+        with:
+          workflow: benchmark.yml
+          workflow_conclusion: success
+          name: proving_benchmark
+          path: ./
+          if_no_artifact_found: ignore
+
+      - name: Run the script
+        run: |
+          echo "Benchmarking proving with $BENCHMARK_WITNESS"
+          ./scripts/benchmark_input.sh $BENCHMARK_WITNESS | tee benchmark_output.log
+          MEASURED_PROVING_TIME_SEC=`cat benchmark_output.log | grep 'Proving duration:' | tail -1 | awk '{ print $3}'`
+          PERF_TIME=`cat output.log | grep "seconds time elapsed" |  tail -1 | awk '{ print ($1)}'`
+          PERF_USER_TIME=`cat output.log | grep "seconds user" |  tail -1 | awk '{ print ($1)}'`
+          PERF_SYS_TIME=`cat output.log | grep "seconds sys" |  tail -1 | awk '{ print ($1)}'`
+          echo "RUN_ID=${{ github.run_id }} MEASURED_PROVING_TIME_SEC=$MEASURED_PROVING_TIME_SEC \
+            PERF_TIME=$PERF_TIME PERF_USER_TIME=$PERF_USER_TIME \
+            PERF_SYS_TIME=$PERF_SYS_TIME FILE=$BENCHMARK_WITNESS"
+          printf '%12s     %-12s     %-24s    %-20s    %-20s    %-20s    %-s\n' \
+            `date --utc +%y-%m-%d-%H:%M:%S` ${{ github.run_id }} \
+            $MEASURED_PROVING_TIME_SEC $PERF_TIME $PERF_USER_TIME \
+            $PERF_SYS_TIME $BENCHMARK_WITNESS >> proving_benchmark_results.txt
```

Comment:
I just don't think this method of benchmarking is useful.

Suppose I see a 5% regression between one week and the next - how do I know if it's statistically significant or just down to chance?
We're doing an n=1 run every week, which gives us no statistical power.
Every week we run (say) a different codebase, so it's not really sound to use consecutive weeks to get statistical power.

That is why all benchmarking suites use multiple runs to actually measure deltas.

Reply:
@0xaatif We are running it every night. From current tests, they are few hunderd miliiseconds different for block `19807080` on 300 seconds test. Meaning, 5% will be very observable, and we will be able to track the regression to the PR merged in that day.
I think this is quite useful.

> That is why all benchmarking suites use multiple runs to actually measure deltas.

Happy to work further to improve benchmarks. We need to start from somewhere. Now we have nothing.

Reply:
Thanks for the explanation, let's go with this as an initial draft then, though I think we can do better with our timing output :D